### PR TITLE
update and improve merge_service_public_FR

### PR DESCRIPTION
**Pull request by nlehuby**
Following up on https://github.com/osm-fr/osmose-backend/pull/1014#issuecomment-720695681, this PR:
* update the csv dataset (with a new python script to do it) and make according changes in the analysis (mostly improve naming of the columns)
* remove false positives (increase merge distance & add multi-polygon relations support)
* add contact info when available

### CODE REVIEW

**Comment by frodrigo** [comment]
Thank you for looking at this.

Nevertheless the parser can now be done a file format parser. There is no example of this, yet. But must be done by sub-classing Parser https://github.com/osm-fr/osmose-backend/blob/master/analysers/Analyser_Merge.py#L390
So automatic regular data update could be done.

**Comment by nlehuby** [comment]
Ok understood.

Can we still merge this PR ? Even if it does not handle the automatic data update, it should already be better.
And I would like to see if this is "better enough". If not, I may want to improve the geocoding, and I may do it outside of Osmose (if so, as a side effect, the specific data parser won't be needed in Osmose anymore)

**Comment by frodrigo** [comment]
> Can we still merge this PR ? Even if it does not handle the automatic data update, it should already be better.

OK

> And I would like to see if this is "better enough". If not, I may want to improve the geocoding, and I may do it outside of Osmose (if so, as a side effect, the specific data parser won't be needed in Osmose anymore)

Osmose can also do the geocoding (for now, only on CSV https://github.com/osm-fr/osmose-backend/blob/master/analysers/Analyser_Merge_Geocode_Addok_CSV.py )

File: analysers/analyser_merge_service_public_FR.py [hunk 1]
```diff
@@ -36,30 +41,34 @@ def __init__(self, config, logger, item, clas, level, select, osmTags, defaultTa
         self.init(
             u"https://www.data.gouv.fr/fr/datasets/service-public-fr-annuaire-de-l-administration-base-de-donnees-locales/",
             "Service-Public.fr",
-            CSV(Source(attribution = u"Service-Public.fr", millesime = "08/2018",
-                    file = "service_public_FR.csv.bz2"),
-                csv = False, separator = None),
-            Load("lon", "lat",
-                select = {"pivot": select},
+            CSV(Source(attribution = u"Service-Public.fr", millesime = "11/2020",
```
Comment (frodrigo) [inline analysers/analyser_merge_service_public_FR.py:44]:
```suggestion
            CSV(Source(attribution = "Service-Public.fr", millesime = "11/2020",
```

File: analysers/analyser_merge_service_public_FR.py [hunk 2]
```diff
@@ -36,30 +41,34 @@ def __init__(self, config, logger, item, clas, level, select, osmTags, defaultTa
...
                     tags = osmTags),
-                conflationDistance = 300,
+                conflationDistance = 500,
                 generate = Generate(
                     static1 = defaultTag,
                     static2 = {"source": self.source},
-                    mapping1 = dict({"wheelchair": lambda res: self.accTable[res["acc"]] if res["acc"] else None}, **defaultTagMapping),
-                    text = lambda tags, fields: {"en": "{0}, {1} (geocoded {2})".format(fields["nom"], fields["adresse"], self.prescitionTableEn[fields["precision"]]), "fr": "{0}, {1} (géocodé {2})".format(fields["nom"], fields["adresse"], self.prescitionTableFr[fields["precision"]])} )))
+                    mapping1 = dict({"wheelchair": lambda res: self.accTable[res["wheelchair_access"]] if res["wheelchair_access"] else None,
+                        "contact:email":lambda res:res["email"] if "@" in res["email"] else None,
```
Comment (frodrigo) [inline analysers/analyser_merge_service_public_FR.py:68]:
```suggestion
                        "contact:email":l ambda res:res["email"] if "@" in res["email"] else None,
```

File: merge_data/service_public_FR.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+# coding: utf-8
+
+import csv
+import os
+import xml.etree.ElementTree as ET
+
+path = 'organismes/'
+
+def get_poi_info(file_name):
+    print(file_name)
```
Comment (frodrigo) [inline merge_data/service_public_FR.py:11]:
```suggestion
```

File: merge_data/service_public_FR.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+organismes = []
+
+for r, d, f in os.walk(path):
+    for file_ in f:
+        if '.xml' in file_:
+            file_name = os.path.join(r, file_)
+            organismes.append(get_poi_info(file_name))
+
+with open("service_public_FR.csv", 'w') as out_file:
+    wr = csv.DictWriter(out_file, quoting=csv.QUOTE_ALL, fieldnames = organismes[0].keys())
```
Comment (frodrigo) [inline merge_data/service_public_FR.py:61]:
```suggestion
    wr = csv.DictWriter(out_file, quoting=csv.QUOTE_ALL, fieldnames=organismes[0].keys())
```

File: analysers/analyser_merge_service_public_FR.py [hunk 5]
```diff
@@ -36,30 +41,34 @@ def __init__(self, config, logger, item, clas, level, select, osmTags, defaultTa
...
-                    precision VARCHAR(254)"""),
+                    category VARCHAR(254),
+                    name VARCHAR(254),
+                    email VARCHAR(254),
+                    phone VARCHAR(254),
+                    wheelchair_access VARCHAR(254),
+                    latitude VARCHAR(254),
+                    longitude VARCHAR(254),
+                    geoloc_precision VARCHAR(254),
+                    address VARCHAR(1024)"""),
```
Comment (frodrigo) [inline analysers/analyser_merge_service_public_FR.py:58]: Since you switch to std CSV, I think you can remove this.

**Comment by frodrigo** [comment]
Merged. Thank you!

**Comment by frodrigo** [comment]
Deployed and updated.

**Comment by nlehuby** [comment]
pfff :disappointed: 
image

**Pull request merged**